interfaces/matlab_experimental/Utility/cleanup.m

```diff
@@ -0,0 +1,12 @@
+function cleanup()
+    % Delete all stored Cantera objects and reclaim memory.
+    %
+    checklib;
+    callct('ct_clearOneDim');
+    callct('ct_clearMix');
+    callct('ct_clearFunc');
+    callct('ct_clearStorage');
+    callct('ct_clearReactors');
+    callct('ct_clearReactionPath');
+    clear all
```

Comment:
not sure that this should call `clear all` - function should be specific to `Cantera`; if a user wants to create their own shortcut, they are free to do so.

Reply:
Yeah that's right. Another thing is the user should always call `clear` on Cantera objects before running `cleanup`, otherwise it would crash Matlab. Would it be helpful to add a fail-safe mechanism to scan in the Workspace for undeleted Cantera objects and give the user a warning?

Reply:
> Would it be helpful to add a fail-safe mechanism to scan in the Workspace for undeleted Cantera objects and give the user a warning?

I think that cleanup should automatically detect and delete Cantera objects, but leave other variables untouched.

Reply:
> > Would it be helpful to add a fail-safe mechanism to scan in the Workspace for undeleted Cantera objects and give the user a warning?
> 
> I think that cleanup should automatically detect and delete Cantera objects, but leave other variables untouched.

I'm still thinking of a way to clear certain variables in the base workspace from a function in Matlab. 

Another thing is, since we already implemented class destructors, is it necessary to use those `ct_clear` methods at all? All objects in the workspace with a linked C++ object will be deleted and Matlab cleans up any temporary objects created by functions, so I don't think there will be loose ends that need clearing. Currently Cantera would throw the following error whenever I call `Cantera.cleanUp`:
```
*******************************************************************************
CanteraError thrown by Cabinet::del:
Attempt made to delete an already-deleted object.
*******************************************************************************
```
This would stop `Cantear.cleanUp` from completion. I think this should be a warning instead.

Reply:
Thanks for pointing out the destructors. If these work as intended, is there any need for `Cantera.cleanup` (or to call the underlying `ct_clear` methods)? The one thing I am not sure about is what happens when a user calls `Cantera.unload` - in this case variables would still exist and a cleanup is necessary? What happens if a user shuts down MATLAB and Cantera is still loaded? Hopefully both can be handled without a dedicated cleanup command while avoiding hard crashes or segfaults?

Reply:
Sorry I saw this comment late. But yes I did implement a new cleanup method that scans for all Cantera objects in the workspace and call destructors on them, and only then will ctUnload unloads Cantera.